packages/core/test/event.test.ts

```diff
@@ -26,39 +27,34 @@ describe('@bugsnag/core/event', () => {
 
   describe('addMetadata()', () => {
     it('updates a whole new section', () => {
-      const Event = require('../event')
       const r = new Event('Err', 'bad', [])
       r.addMetadata('specific detail', { extra: 'stuff' })
-      expect(r._metadata['specific detail']).toEqual({ extra: 'stuff' })
+      expect((r._metadata as any)['specific detail']).toEqual({ extra: 'stuff' })
```

Comment:
Here's an example where the types inferred by TS from the JS file are insufficient. In this case it infers _metadata to have type `{}` so casting has been used to hide the type error. A better way might be to have a sibling `event.d.ts` file that properly defines the types so that we don't lose the benefits of static type checking. The problem is how to reconcile that with the other `event.d.ts` that represents only the API we wish consumers to use.